docs/madr/decisions/020-kds-redesign.md

```diff
@@ -0,0 +1,238 @@
+
+### Create resources on the zone
+
+We intend to introduce a new feature that enables the creation of resources at the zone level. These resources will be exclusively applicable to the specific zone, and we will synchronize them to the global level as read-only. To distinguish the zone resources from global ones, we will prefix them with the name of the zone.
+
+```
+zone-1.example-timeout
+```
+
+### Resources
+The following resources are the ones we should permit creation in the zone:
+* CircuitBreaker
+* ExternalService
+* FaultInjection
+* HealthCheck
+* MeshGateway
+* MeshGatewayRoute
+* ProxyTemplate
+* RateLimit
+* Retry
+* Timeout
+* TrafficLog
+* TrafficPermission
+* TrafficRoute
+* TrafficTrace
+* VirtualOutbound
+* MeshAccessLog
+* MeshCircuitBreaker
+* MeshFaultInjection
+* MeshHealthCheck
+* MeshHttpRoute
+* MeshProxyPatch
+* MeshRateLimit
+* MeshRetry
+* MeshTimeout
+* MeshTrace
+* MeshTrafficPermission
+
+When there are two policies, one at the zone level and another at the global level, that impact the same dataplanes, the more specific policy takes precedence. In this scenario, the Zone policy would be the one utilized.
+
+Global policy:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+ name: timeout-global
+ namespace: kuma-system
+ labels:
+   kuma.io/mesh: default
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+   - targetRef:
+       kind: Mesh
+     default:
+       idleTimeout: 20s
+       connectionTimeout: 2s
+```
+
+Zone policy:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+ name: timeout-zone
+ namespace: kuma-system
+ labels:
+   kuma.io/mesh: default
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+   - targetRef:
+       kind: Mesh
+     default:
+       idleTimeout: 10s
+       connectionTimeout: 2s
+```
+
+The policy `timeout-zone` is going to be applied to all dataplanes in the default mesh in the zone.
+
+### Incremental xDS
+
+Incremental xDS in the `DeltaDiscoveryRequest` has a field `resource_names_subscribe` which requires information about resources that we want to subscribe to e.g. policy with name `timeout-1`. If you send `resource_names_subscribe:  "*"` in a `DeltaDiscoveryRequest`, the control plane should respond with all resources of the specified type, not just the ones that have changed since the last update. This is because the `resource_names_subscribe` field is used to filter which resources the control plane should send, and if it is set to `*`, it effectively disables filtering and requests all resources of the specified type.
+
+xDS Specification
+* https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#how-the-client-specifies-what-resources-to-return
+* https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#incremental-xds
+* https://github.com/envoyproxy/go-control-plane/blob/main/pkg/cache/v3/delta.go#L31
```

Comment:
On our side what will be needed here? Do we need to write an XDS Delta client?

Reply:
Yes, it seems like go-control-plane has no client and we might have to write one.